WorkloadManagementSystem/Service/JobStateUpdateHandler.py

```diff
@@ -183,61 +181,51 @@ def export_setJobStatusBulk(self, jobID, statusDict):
-      sDict = statusDict[date]
-      if sDict['Status']:
-        status = sDict['Status']
+    if dates[-1] >= lastTime:
+      # Get the last status values
+      for date in [date for date in dates if date >= lastTime]:
+        sDict = statusDict[date]
+        status = sDict.get('Status', status)
         if status in JobStatus.JOB_FINAL_STATES:
           endDate = date
-        if status == JobStatus.RUNNING:
-          startFlag = JobStatus.RUNNING
-      if sDict['MinorStatus']:
-        minor = sDict['MinorStatus']
-        if minor == "Application" and startFlag == JobStatus.RUNNING:
+        minor = sDict.get('MinorStatus', minor)
+        # Pick up the start date
+        if minor == "Application" and status == JobStatus.RUNNING and not startDate:
           startDate = date
-      if sDict['ApplicationStatus']:
-        application = sDict['ApplicationStatus']
-      counter = sDict.get('ApplicationCounter')
-      if counter:
-        appCounter = counter
-    attrNames = []
-    attrValues = []
-    if status:
-      attrNames.append('Status')
-      attrValues.append(status)
-    if minor:
-      attrNames.append('MinorStatus')
-      attrValues.append(minor)
-    if application:
-      attrNames.append('ApplicationStatus')
-      attrValues.append(application)
-    if appCounter:
-      attrNames.append('ApplicationCounter')
-      attrValues.append(appCounter)
-    result = jobDB.setJobAttributes(jobID, attrNames, attrValues, update=True)
-    if not result['OK']:
-      return result
+        application = sDict.get('ApplicationStatus', application)
+        appCounter = sDict.get('ApplicationCounter', appCounter)
+
+      attrNames = []
+      attrValues = []
+      if status:
+        attrNames.append('Status')
+        attrValues.append(status)
+      if minor:
+        attrNames.append('MinorStatus')
+        attrValues.append(minor)
+      if application:
+        attrNames.append('ApplicationStatus')
+        attrValues.append(application)
+      if appCounter:
+        attrNames.append('ApplicationCounter')
+        attrValues.append(appCounter)
+      result = jobDB.setJobAttributes(jobID, attrNames, attrValues, update=True)
+      if not result['OK']:
+        return result
 
-    if endDate:
-      result = jobDB.setEndExecTime(jobID, endDate)
-    if startDate:
-      result = jobDB.setStartExecTime(jobID, startDate)
+      if endDate:
+        result = jobDB.setEndExecTime(jobID, endDate)
+      if startDate:
+        result = jobDB.setStartExecTime(jobID, startDate)
 
     # Update the JobLoggingDB records
     for date in dates:
       sDict = statusDict[date]
-      status = sDict['Status']
-      if not status:
-        status = 'idem'
-      minor = sDict['MinorStatus']
-      if not minor:
-        minor = 'idem'
-      application = sDict['ApplicationStatus']
-      if not application:
-        application = 'idem'
+      status = sDict['Status'] if sDict['Status'] else 'idem'
```

Comment:
```suggestion
      status = sDict.get(Status', 'idem')
```

Reply:
Unfortunately it is not the same... Usually `'Status'` is in the dictionary but as an empty string!

Reply:
Good to know! Should be commented then

Reply:
`get('Status','idem')` would only work if there would be no key `Status`. Here the code assumed there is always this key, which is true, but for a strange reason the logging info relies on this string `'idem'` while it could just take an empty string and consider this means the status is unchanged! Probably historical from developers who were not too sure...